firefox-ios/firefox-ios-tests/Tests/ClientTests/Fakespot/FakespotViewModelTests.swift

```diff
@@ -8,30 +8,113 @@ import XCTest
 final class FakespotViewModelTests: XCTestCase {
     var client: MockShoppingProduct!
     var apiClient: TestFakespotClient!
+    var productAdsCache: ProductAdsCache!
 
     override func setUp() {
         super.setUp()
         DependencyHelperMock().bootstrapDependencies()
         apiClient = TestFakespotClient()
         client = MockShoppingProduct(url: URL(string: "https://www.example.com")!, client: apiClient)
+        productAdsCache = .shared
     }
 
     override func tearDown() {
         super.tearDown()
         client = nil
         apiClient = nil
+        productAdsCache.clearCache()
+        productAdsCache = nil
     }
 
     // Mock ShoppingProduct for testing
     class MockShoppingProduct: ShoppingProduct {
         var shouldThrowError = false
+        var shouldFetchAds = true
+        // Mocked ads data for testing
+        let ads = [
+            ProductAdsResponse(
+                name: "Mocked Ads",
+                url: URL(string: "https://mocked-ads.com")!,
+                imageUrl: URL(string: "https://mocked-ads.com/image.jpg")!,
+                price: "29.99",
+                currency: "USD",
+                grade: .a,
+                adjustedRating: 4.8,
+                analysisUrl: URL(string: "https://mocked-as.com/analysis")!,
+                sponsored: true,
+                aid: "mocked123"
+            )
+        ]
 
         override func fetchProductAnalysisData(maxRetries: Int = 3, retryTimeout: Int = 100) async throws -> ProductAnalysisResponse? {
             if shouldThrowError {
                 throw NSError(domain: "MockErrorDomain", code: 123, userInfo: nil)
             }
             return .empty
         }
+
+        override func fetchProductAdsData() async -> [ProductAdsResponse] {
```

Comment:
This boolean is not really concludent way of telling us if we should fetch ads or not. Ideally we would mock prefs and the cache and not override this method.

Reply:
Perhaps there's confusion about the naming. I aim to simulate server responses, both with data and without data, by overriding the method.